### PR TITLE
Updated some older sets, added some Blizzard Store sets

### PR DESCRIPTION
**Pull request by mikiurban**

### CODE REVIEW

**Comment by mikiurban** [comment]
I have to admit, my LUA Is horrible.  My entire professional career in lua has been working on this addon.  Anyways, I added 3 things besides the normal set updates:

1. I added "function SetCollector:CreateAppearanceFromItemID(itemID)" which is similar to your A() macro, but instead of taking an Appearance ID and Source ID, just takes an Item ID.  It should make hand-making custom sets easier in the future, you just need to know the Item IDs, which is a whole lot easier.  You can see I changed one set in 01.lua.
2. I added a table called "setMap" to the global table, generated by the world's ugliest LUA code.  There should be a legal waiver before you are allowed to read it.  It makes a map of each set ID, their base set ID and collection description (RAID, PVP, OTHER, etc).  Eventually, I think it would be cool to use this as the "source of truth" instead of the 01-10.lua files.  The addon can iterate through every set in game (C_TransmogSets.GetAllSets()) then use that map to determine which collection they should appear.  New, unconfigured sets could go in "Unsorted" or something, until we update the map and put them in their correct place.  The BaseSetId would be to override the default base set id, like when a base set is a raid set but the variant is a Darkmoon Faire vendor item.  Homemade sets would still need a DB file, but we could get rid of about 90% of the 00.lua thru 10.lua data files.
3. But for now, that setMap powers "/sc export" which uses the world's second ugliest LUA, right behind the setMap builder, to auto-generate the entire IncludeSet() command, including variants, for every set in the entire game.  Run it and /reload and check out your SavedVariables file under ["export"].  The sort-order param is wrong (I could prob put that in setMap, but didn't think of it until now), but I also generate the set names and variants in comments for each line.  You can see which sets are not defined in any of the ##.lua files by searching for "SetCollector.(missing)", and as of this pull request, the only missing ones are Heritage Armor and some placeholder things.

**Comment by mikiurban** [comment]
It's late and my wife is telling me to go to bed, so I prob forgot to mention something. oh, well, see ya later.

**Comment by mikiurban** [comment]
Added The Forbidden Reach variant of Primal Elements set from Patch 10.0.7

File: Appearances/Base.lua [hunk 1]
```diff
@@ -96,11 +110,16 @@ function SetCollector:IncludeSet(collection, uid, setID, armorType, class, facti
             Location = description,
             Variants = { SetCollector:IncludeVariant(setID, setInfo) }
         }
+
+        SetCollector.db.global.setMap["SET " .. setID] = {setID = setID, collection = collection.Description, baseSetId = setID}
```
Comment (scottolsonjr) [inline Appearances/Base.lua:114]: I like the concept. I have been wanting to explore a way to provide additional search/filtering options around the existing Blizzard defined sets while allowing for Set Collector and user-defined sets.

Comment (scottolsonjr) [reply Appearances/Base.lua:114]: We'll want to keep an eye on memory usage. I have the expansions split out allowing for users to exclude expansions they don't care about to keep memory usage down as the game introduces more expansions and sets

File: Core.lua [hunk 2]
```diff
@@ -337,9 +345,36 @@ function ParseClassMask(bits)
...
+    local parse = bits .. ": "
+    for char, mask in pairs(map) do
+        if (BitAND(bits, mask) == mask) then
+            parse = parse .. char .. " "
+        end
+    end
+    return parse .. ""
+end
+
 function SetCollector:ExportSetData()
```
Comment (scottolsonjr) [inline Core.lua:375]: In a future update, let's get the export code moved into its own lua file

**Review by scottolsonjr — APPROVED** [review]

File: Appearances/Base.lua (hunk 1 above)
Comment (mikiurban) [reply Appearances/Base.lua:114]: > We'll want to keep an eye on memory usage. I have the expansions split out allowing for users to exclude expansions they don't care about to keep memory usage down as the game introduces more expansions and sets

Yeah, the memory usage has been on the backburner, my code is horrible about it right now as I try to figure out what info is important.  Also, if I knew what I actually wanted to do, that would help me figure out what info is important.  I think I'm on "Plan F" now, after a couple false starts and realizing plans D and E didn't go far enough.

File: Core.lua (hunk 2 above)
Comment (mikiurban) [reply Core.lua:375]: > In a future update, let's get the export code moved into its own lua file

Yeah, I wanted you to have the updated set info, and didn't have enough time to actually be a good developer, hehe.  I kinda meant for that code to not be included :/

File: Appearances/Base.lua (hunk 1 above)
Comment (mikiurban) [reply Appearances/Base.lua:114]: > I like the concept. I have been wanting to explore a way to provide additional search/filtering options around the existing Blizzard defined sets while allowing for Set Collector and user-defined sets.

Yeah.  The Blizzard UI is nice and helpful, but not (afaik) expandable.  Ultimately, I'd like to add a search/filter bar like the Blizzard UI has, and then add all the Ensembles: X in the game that aren't (for some crazy reason) flagged as a cosmetic set.  We're going to need more categories!

**Pull request merged**